mysql/datadog_checks/mysql/activity.py

```diff
@@ -135,7 +135,11 @@ def run_job(self):
             self._log.debug(
                 'Waiting for events_waits_current availability to be determined by the check, skipping run.'
             )
-        if self._check.events_wait_current_enabled is False:
+        # Azure flexible server 
+        azure_deployment_type = ""
+        if self._config.cloud_metadata.get("azure") is not None:
+            azure_deployment_type = self._config.cloud_metadata.get("azure")["deployment_type"]
```

Comment:
```suggestion
        azure_deployment_type = self._config.cloud_metadata.get("azure", {}).get("deployment_type")
```

Reply:
wouldn't I have to do some validation before comparing a string and a possible empty object? or python magic handles it? 🪄

Reply:
if azure_deployment_type is None, azure_deployment_type != 'flexible_server' will still be False.

Reply:
I see, i'm used to having type comparison errors